### PR TITLE
feat: simplify and split models/definition/filters files

### PR DESCRIPTION
**Pull request by buuhuu**
- [x] simplify model files by removing any attributes of the fields that are not strictly required (`valueType`, `value`,  ...)
- [x] rename properties of default content models to adhere to block naming conventions (field collapsing) 
- [x] split models/definitions/filters into partials
- [x] release the used merge-json-cli to npmjs.org

Each block now has a `_component.json` partial with a json object that contains `definitions`, `models` and `filters`. Each of them is an array as a block may expose 0..n of either of them. The default content model partials are now stored in `models/` as well as the partials for component-models.json, component-definition.json and component-filters.json. The file name format with the leading `_` was inspired by sass partials (files that are meant to be imported into other files).

Besides that this commit adds `husky` pre-commit hook support, with a simple hook that merges the models/definition/filters files if any of the partials changed.

### CODE REVIEW

File: models/_component-definition.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
+  "groups": [
+    {
+      "title": "Default Content",
+      "id": "default",
+      "components": [
+        {
+          "...": "./_text.json#/definitions"
```
Comment (sdmcraft) [inline models/_component-definition.json:8]: What does `...` signify here? Is this a way of specifying a reference from where to fetch the definition?

File: blocks/cards/component.json [hunk 2]
```diff
@@ -0,0 +1,63 @@
+{
```
[the remark this answers is not in the document]
Comment (buuhuu) [reply blocks/cards/component.json]: You are right, fixed the name.

File: models/_component-definition.json (hunk 1 above)
Comment (buuhuu) [reply models/_component-definition.json:8]: The `...` is similar to the javascript spread operator. See https://www.npmjs.com/package/merge-json-cli

**Review by msagolj — APPROVED** [review]

**Comment by jckautzmann** [comment]
A few suggestions:
- I’d rename `blocks/cards/_component.json` to `blocks/cards/_cards.json` to be consistent with `models/_button.json`
- I’d rename `component-definition.json` and `_component-definition.json` to make it plural: `component-definitions.json` and `_component-definitions.json` to make it consistent with `component-filters` and `component models`

**Comment by buuhuu** [comment]
@jckautzmann 
> I’d rename blocks/cards/_component.json to blocks/cards/_cards.json to be consistent with models/_button.json

Very good suggestion as this allows 
- (a) to use autocomplete search and find the .json by name. Currently if you search for components you find a lot.
- and (b) allows to split it even further per block like `_cards.json` and `_card.json`

> I’d rename component-definition.json and _component-definition.json
that would be a breaking change which I would not want to do now. I agree though.

**Comment by clementepereyra** [comment]
Hi all,
Any ETA when this will be merged? I'm about to migrate a project and would like to know if I should wait a little bit, or just move forward with the current approach.
Thanks.

**Comment by buuhuu** [comment]
This will be merged after the next AEM Cloud Services release scheduled for this / next week.

The majority of changes can be used. Unfortunately I made some changes to the default content models (button, title, image) and rename the fields according to our naming conventions so that they can be used in container blocks without any changes. This change is the only thing that requires the AEM CS release. Spitting/merging the files works today.

**Pull request merged**